packages/at_commons/lib/src/keystore/at_key.dart

```diff
@@ -48,6 +48,9 @@ class AtKey {
 
   set sharedWith(String? atSign) {
     assertStartsWithAtIfNotEmpty(atSign);
+    if(isLocal == true || metadata?.isPublic == true){
```

Comment:
When the supplied atSign parameter is null, the checks for isLocal and isPublic should not be made.

Reply:
Please write a test which first sets isLocal to true and then sets sharedWith.

Reply:
@gkc: I am not clear on the first comment. If the comment is meant to skip the `IF` condition when atSign is null, then`assertStartsWithAtIfNotEmpty` method throws an exception if the atSign is null. So, the subsequent check will not execute.

Added an unit test where isLocal to set to true first and then sharedWith is added.

Reply:
assertStartsWithAtIfNotEmpty will not throw an exception if the supplied value is null. so the subsequent check will indeed execute

Reply:
please write a test where you are supplying a null sharedWith

Reply:
Added a unit test and also refactored code when sharedWith is set to null on a local key or a public key